rpc/README.md

```diff
@@ -0,0 +1,96 @@
+# RPC
+
+**RPC** is an externally-facing component through which clients can interact with the node. It receives client requests 
+(e.g., to synchronize with the latest state of the chain, or to submit transactions), performs basic validation, 
+and forwards the requests to the appropriate internal components.
+**RPC** is one of components of the [Miden node](..).
+
+## Architecture
+
+The Miden node is still under heavy development and current architecture is subject of change. This topic will be 
+filled later.
+
+## Usage
+
+### Installing RPC
+
+RPC can be installed and run as a part of [Miden node](../README.md#installing-the-node).
+But if you intend on running RPC as a separated process, you need to install and run it separately:
+
+```sh
+# Installs `miden-node-rpc` executable
+cargo install --path rpc
+```
+
+To run the RPC you'll need to provide a configuration file. We have an example config file in [rpc-example.toml](rpc-example.toml).
+
+Then, to run an RPC:
+
+```sh
+miden-node-rpc serve --config <path-to-rpc-config-file>
+```
+
+## API
+
+**RPC** serves connections using [gRPC protocol](https://grpc.io) on a port, set in configuration file. Here is a brief 
+description of supported methods.
+
+### CheckNullifiers
+
+Gets a list of proofs for given nullifier hashes, each proof as Tiered Sparse Merkle Trees ([read more](../proto/proto/tsmt.proto)). 
+
+**Parameters:**
+
+* `nullifiers`: `[Digest]` – array of nullifier hashes.
+
+**Returns:**
+
+* `proofs`: `[NullifierProof]` – array of nullifier proofs, positions correspond to the ones in request.
+
+### GetBlockHeaderByNumber
+
+Retrieves block header by given block number.
+
+**Parameters**
+
+* `block_num`: `uint32` *(optional)* – the block number of the target block. If not provided, the latest known block will be returned.
+
+**Returns:**
+
+* `block_header`: `BlockHeader` – block header.
+
+### SyncState
+
+State synchronization request.
```

Comment:
I would write a small paragraph on how this is intended to be used as it may not be super obvious. There is some background info in #38 and #43, but I also kind of summarize it in https://github.com/0xPolygonMiden/miden-client/pull/63#discussion_r1432583783.